src/compiler/scanner.ts

```diff
@@ -285,9 +285,13 @@ namespace ts {
     }
 
     const tokenStrings = makeReverseMap(textToToken);
+    // "this" maps to SyntaxKind.ThisKeyword normally, so can't be in textToToken. But can put it in tokenStrings.
+    tokenStrings[SyntaxKind.ThisType] = "this";
 
     export function tokenToString(t: SyntaxKind): string | undefined {
-        return tokenStrings[t];
+        const res = tokenStrings[t];
+        Debug.assert(!!res);

```

Comment:
That's a breaking change since it used to work with any SyntaxKind before, e.g. `StringLiteral` which is a token and should therefore be a valid input. 

Otherwise you can change the return type and remove the undefined there.

Reply:
Oh well, better to avoid the breaking change.